Add summary of capabilities/user stories to docs index page (#149)
astropy/specreduce: docs/index.rst

Specreduce Documentation
########################

The `specreduce <https://specreduce.readthedocs.io/en/latest/index.html>`_ package
aims to provide a data reduction toolkit for optical
and infrared spectroscopy, on which applications such as pipeline processes for
specific instruments can be built. The scope of its functionality is limited to
basic spectroscopic reduction, currently encompassing the following three tasks:

#. Determining the trace of a spectrum dispersed in a 2D image, either by setting a flat

>    trace or fitting a spline, polynomial, or other model to the positions of the dispersed

Reviewer: I would also add that users can set a custom trace so `ArrayTrace` isn't left out.

   spectrum.
#. Generating a background based on a region on one or both sides of this trace, and making
   available the background image, 1D spectrum of the background, and the
   background-subtracted image.

> #. Performing either a Horne (aka "optimal") or boxcar extraction on either the original

Reviewer: ```suggestion
#. Performing either a Horne (a.k.a. "optimal") or boxcar extraction on either the original
```

   or background-subtracted 2D spectrum, using the trace generated by the first task to
   generate a 1D spectrum.

Further details about these capabilities are detailed in the sections linked below.
Beyond these tasks, basic *image* processing steps (such as bias subtraction) are covered by
`ccdproc <https://ccdproc.readthedocs.io/en/latest/>`_
and other packages, data analysis by `specutils <https://specutils.readthedocs.io/en/latest/>`_,
and visualization by `matplotlib <https://matplotlib.org/>`_. A few

> examples will be provided of using ``specreduce`` in conjunction with these

Reviewer: ```suggestion
examples will be provided that feature ``specreduce`` in conjunction with these
```
